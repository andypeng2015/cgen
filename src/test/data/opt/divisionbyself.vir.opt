module foo

export function w $foo(w %x) {
@2:
  %4 = ne.w %x, 0x0_w ; @11
  %5 = flag.w %4 ; @12
  ret %5
}
