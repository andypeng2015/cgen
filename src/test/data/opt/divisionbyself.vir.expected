module foo

export function w $foo(w %x) {
@0:
  %0 = eq.w %x, 0x0_w ; @5
  br %0, @1, @2
@1:
  ret 0x0_w
@2:
  ret 0x1_w
}
